### ISSUE TITLE
Simple Comparison of Generics not allowed?

### ISSUE THREAD

**Issue by bradennapier** [microsoft/TypeScript#38220]
**TypeScript Version:**  3.8

**Search Terms:**
generic comparison
compare generic strings

**Code**

```ts
function test<A extends string, B extends string>(one: A, two: B) {
  // This condition will always return 'false' since the types 'A' and 'B' have no overlap.ts(2367)
  if (one === two) {
    throw new Error('no');
  }
}

test('one', 'one');
```

**Expected behavior:**

While the generics are generally indicating that `one` and `two` are independent values, they can still potentially be the same underlying value so they should be able to at least be checked for equality without casting them first, especially since there is no error produced when calling the function with identical values in the first place.

At the very least the error message should be more specific since the assumption it is providing is not correct in since the condition returns true :-P

**Actual behavior:**

Shown in example

**Playground Link:** [Playground](https://www.typescriptlang.org/v2/en/play?ts=3.8.3#code/GYVwdgxgLglg9mABFApgZygHgIKJQD1TABM1EMAnGMAcwBpEAhPQlEsy6mgPgAoEUALkTYGUAO5xhjAJTDcAHyaIA3gFgAUIkQxgifmBSIAvKeSSZqzdu1QAFhTjjEh5wFEKjirwDkYOD4yANzWiAC+oRQoUCAUSAIhGhEamqgYvgI+DD6ZwUA)
